packages/x-components/src/components/animations/animation.abstract-factory.ts

```diff
@@ -0,0 +1,33 @@
+import Vue, { VueConstructor } from 'vue';
+
+/**
+ * Abstract Factory to create animations Factory. The returned animation factory uses the
+ * `animationName` parameter to create a Transition Component with that name.
+ *
+ * @param animationName - The name to use in the Transition Component of animation.
+ * @returns The animation factory configured.
+ *
+ * @internal
+ */
+export function animationAbstractFactory(
```

Comment:
I know some times might be good to add in to the name of what you are doing the pattern, but wouldn't it be more natural to just call it  something more explicit and easier like `createDirectionalAnimationFactory`. We even have this in our naming guide:

>
> ## Methods and functions
> 
> - Methods and functions must start with a verb.